src/cmd/troubleshoot/image_pullable.go

```diff
@@ -155,6 +152,64 @@ func checkComponentImagePullable(httpClient *http.Client, componentName string,
 	return nil
 }
 
+func checkCustomModuleImagePullable(httpClient *http.Client, componentName string, pullSecret string, componentImage string) error {
```

Comment:
why is this needed if we have `checkComponentImagePullable` ?
what is the difference ?
AFAIK, the `codeModulesImage` is just as an image, just like the `image` field.
The content of the image is different, but the "is it pullable" check should be the same for both.

Reply:
The URL format of the component images are specific to DT, whereas custom images can have any URL. So they need different treatment.

Reply:
AFAIK, the `image` field can also have any kind of URL

Reply:
The function `checkOneAgentImagePullable` gets the right image which needs to be pulled from `getOneAgentImageEndpoint` this can return an image that is either from our own registry like tenant.dynatrace.com/v2/linux/oneagent:latest or an custom image if its set (see the if statements in that function) this could be like quay.io/myfanceregistry/repo/agent:something. This output is passed to `checkComponentImagePullable` which does not care where the image is (either own registry our Dynatrace one) The same can also be done with the codeModules image.